### ISSUE TITLE
`else if(...) {` supported on its own line, but not `else {`?

### ISSUE THREAD

**Issue by sam-github** [eslint/eslint#1849]
```
    else if (...) {                           
      // ...
    }                                                                            
    else {                                                                       
      // ...
      break;                                                                     
    }                                     
```

I can't figure out how get rid of `132:9  error  Closing curly brace does not appear on the same line as the subsequent block  brace-style`, other than wrapping else up to the preceeding `}`, unlike all the else-if statements.

Bug, infelicity, or am I missing a config option?

**Comment by nzakas** [comment]
What did you configure for `brace-style`?

**Comment by sam-github** [comment]
https://github.com/strongloop/strong-supervisor/blob/try-linters/.eslintrc#L96: '1tbs'

So, I can see why it doesn't like the `else`, but I don't know why it didn't warn about the dozen line-leading `else if` before it: https://github.com/strongloop/strong-supervisor/blob/try-linters/lib/options.js#L126-L134

I'll remove the brace-style option, thank you.

**Comment by ilyavolodin** [comment]
The reason why it didn't complain about `else if` is because `else` block is treated differently in `brace-style` rule then `else if`. There is probably an bug in the code that treats `else if` blocks here: https://github.com/eslint/eslint/blob/master/lib/rules/brace-style.js#L91

**Comment by nzakas** [comment]
You're right, looks like a bug.

**Comment by nzakas** [comment]
Working on this now.

**Comment by nzakas** [comment]
Hmm, I'm not sure there's a bug here. I tried the following code:

```js
    if (option === '-h') {
      options.help = true;
    }
    else if (option === '--help') {
      options.help = true;
    }
```

and I correctly got an error flagged on line 4.

What version of ESLint are you using? Is there any other information you can provide? At the moment, I'm stumped.

**Comment by sam-github** [comment]
@nzakas  see my [comment](https://github.com/eslint/eslint/issues/1849#issuecomment-75171571), the repo and branch I'm running eslint on is public, and the .eslintrc is checked in.

I upgrade eslint before checking the old version, sorry, and it was somewhat out of date.

However, now I can't run eslint at all!

```
sam@samtu:~/w/sn/strong-supervisor (try-linters *) % eslint bin lib

/usr/local/lib/node_modules/eslint/lib/rules/consistent-return.js:55
                returnTypeDefined = "type" in returnInfo;
                                              ^
TypeError: Cannot use 'in' operator to search for 'type' in undefined
    at EventEmitter.ReturnStatement (/usr/local/lib/node_modules/eslint/lib/rules/consistent-return.js:55:47)
    at EventEmitter.emit (events.js:117:20)
    at Controller.controller.traverse.enter (/usr/local/lib/node_modules/eslint/lib/eslint.js:683:25)
    at Controller.__execute (/usr/local/lib/node_modules/eslint/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/usr/local/lib/node_modules/eslint/node_modules/estraverse/estraverse.js:495:28)
    at EventEmitter.module.exports.api.verify (/usr/local/lib/node_modules/eslint/lib/eslint.js:676:24)
    at processFile (/usr/local/lib/node_modules/eslint/lib/cli-engine.js:172:23)
    at /usr/local/lib/node_modules/eslint/lib/cli-engine.js:271:26
    at /usr/local/lib/node_modules/eslint/lib/util/traverse.js:61:17
    at Array.forEach (native)
sam@samtu:~/w/sn/strong-supervisor (try-linters *%) % eslint -v
v0.15.0
```

node 0.10.35, fyi.

I see some references to this error in the old issues, but related to arrow functions, which I'm not using.

Can you reproduce this?

**Comment by sam-github** [comment]
One of my co-workers can run eslint, but he's on 0.11.13, I'm trying some newer nodes to see if its related to node version.

**Comment by nzakas** [comment]
Please file a separate issue for that. It's not related to this one.

**Comment by ilyavolodin** [comment]
I just verified the following code:
```js
/* eslint brace-style: [2, "1tbs"] */

var foo = 0;

if (foo) {
	alert("blah");
} else if (!foo) {
	alert("second blah");
} else {
	alert("good");
}
```
in online demo and it's not causing any issues. Closing this. Please reopen if it's still an error in your code.

**Comment by sam-github** [comment]
@ilyavolodin 

the example you tested in https://github.com/eslint/eslint/issues/1849#issuecomment-100397247  isn't related to the bug report, see https://github.com/eslint/eslint/issues/1849#issue-58285459

Rewriting your example:

```
if (foo) {
    alert("blah");
}
else if (!foo) {
    alert("second blah");
}
else {
    alert("good");
}
```

Above is the style I couldn't get eslint to accept, and where the `else if (..) {` were accepted, but not the trailing `else  {`.

**Comment by ilyavolodin** [comment]
@sam-github Sorry, I didn't notice new lines. I just saw that you set brace style to 1tbs and as such I tried 1tbs style. Your example above doesn't confirm to 1tbs style, so it should fails correctly in eslint. Here's modified example:
```js
/* eslint brace-style: [2, "1tbs"] */

var foo = 0;

if (foo) {
    alert("blah");
}
else if (!foo) {
    alert("second blah");
}
else {
    alert("good");
}
```
Fails with the following two errors:
```
8:5 - Closing curly brace does not appear on the same line as the subsequent block.
11:5 - Closing curly brace does not appear on the same line as the subsequent block.
```
So both `else` and `else if` are correctly flagged.

In order to get the code above to pass you should switch to `stroustrup` style for `brace-style` rule.

**Comment by sam-github** [comment]
OK, thank you.

**Comment by paribaranwal** [comment]
Always put last else block at end of its previous block like:
if(){
} else {
}

**Comment by huangjihua** [comment]
else 前面的｝不能和else 同一行

**Comment by abdelmoula** [comment]
I think writing this way should solve the issue : 
if (foo) {
    alert("blah");
} else if (!foo) {
    alert("second blah");
}
